### PR TITLE
add exemplar to native histogram

### LINKED ISSUE #61: How to add exemplars to pure Native Histograms
How to deal with exemplars if there are only sparse buckets but no conventional buckets?

We probably need a separate section in the protobuf spec. (There will also be the interesting question how to pick exemplars, but that's not the concern of the exposition format.)

### PR DESCRIPTION
**Pull request by fatsheep9146**
fix #61

### CODE REVIEW

**Comment by fatsheep9146** [comment]
@beorn7 please help review this pr, thanks!

**Comment by beorn7** [comment]
On my review list.

**Review by beorn7 — CHANGES_REQUESTED** [review]
Thank you. It's an easy change, but I think we should have a comment explaining the context, see my review comment below.

Note that after this is merged, we need to mirror the change in https://github.com/prometheus/prometheus/blob/main/prompb/io/prometheus/client/metrics.proto .

**Comment by beorn7** [comment]
GH ate my review comment. :( I'll write it again.

File: io/prometheus/client/metrics.proto [hunk 1]
```diff
@@ -107,6 +107,8 @@ message Histogram {
   // histograms.
   repeated sint64 positive_delta    = 13; // Count delta of each bucket compared to previous one (or to zero for 1st bucket).
   repeated double positive_count    = 14; // Absolute count of each bucket.
+
+  repeated Exemplar exemplars = 16;
```
Comment (beorn7) [inline io/prometheus/client/metrics.proto:111]: This needs a comment along the lines: "Only used for native histograms. These exemplars MUST have a timestamp."

Comment (fatsheep9146) [reply io/prometheus/client/metrics.proto:112]: Done

**Comment by beorn7** [comment]
Thank you very much. Will you also mirror this into https://github.com/prometheus/prometheus/blob/main/prompb/io/prometheus/client/metrics.proto (and ideally add the code to parse those exemplars and ingest them)?

**Comment by fatsheep9146** [comment]
> Thank you very much. Will you also mirror this into https://github.com/prometheus/prometheus/blob/main/prompb/io/prometheus/client/metrics.proto (and ideally add the code to parse those exemplars and ingest them)?

No problem, I will do this right now @beorn7

**Pull request merged**